libc/src/stdio/baremetal/puts.cpp

```diff
@@ -0,0 +1,25 @@
+//===-- Implementation of puts for baremetal-------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/stdio/puts.h"
+#include "src/__support/CPP/string_view.h"
+#include "src/__support/OSUtil/io.h"
+
+namespace LIBC_NAMESPACE {
+
+LLVM_LIBC_FUNCTION(int, puts, (const char *__restrict str)) {
+  cpp::string_view str_view(str);
+
+  // TODO: Can we combine these to avoid needing two writes?
```

Comment:
I don't think it's possible to combine these without `malloc`ing, unfortunately.

Reply:
The alternative I was thinking about would be to have a `writev` style API where you can provide a vector of strings but I'm not sure if it's worth it.

Reply:
up to you if you want to implement that

Reply:
Not now, maybe as a future improvement.